ckanext/validation/logic.py

```diff
@@ -406,7 +407,8 @@ def _validation_dictize(validation):
     return out
 
 
-def resource_create(context, data_dict):
+@t.chained_action
```

Comment:
how exactly does chained_actions work vs without?

Reply:
Documentation at https://docs.ckan.org/en/2.9/extensions/plugins-toolkit.html#ckan.plugins.toolkit.ckan.plugins.toolkit.chained_action

We provide it as a replacement for `resource_create`, just like usual, but CKAN detects that it's annotated as a chained action, and provides a function reference representing the original function, which we can choose to invoke, or not, potentially with altered parameters.

Multiple chained actions are allowed, and CKAN handles the work of making them call each other in sequence, though I don't think there's any guarantee about ordering.